auth/proofOfPossession_test.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Cloud Security Client Go contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/pem"
+	"github.com/lestrrat-go/jwx/jwt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"os"
+	"path"
+	"testing"
+)
+
+func TestProofOfPossession_ParseCertHeader_edgeCases(t *testing.T) {
+	t.Run("parseCertString() returns nil when no certificate is given", func(t *testing.T) {
+		cert, err := parseCertString("")
+		assert.Nil(t, cert)
+		assert.Nil(t, err)
+	})
+
+	t.Run("parseCertString() fails when DER certificate is corrupt", func(t *testing.T) {
```

Comment:
I think you meant to say base64 encoded instead of DER as DER is binary

Reply:
correct DER are bytes, and "DER Certificate" is meant to be "DER formatted Certificate" - which is quite lengthy...